controllers/k8ssandra/datacenters.go

```diff
@@ -125,23 +125,39 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 				}
 			}
 
-			if !cassandra.DatacenterReady(actualDc) {
-				logger.Info("Waiting for datacenter to become ready")
+			if actualDc.Status.CassandraOperatorProgress != cassdcapi.ProgressReady {
```

Comment:
But we were already doing this. Prior to this change we would call `DatacenterReady` here. It is implemented as follows:

```go
func DatacenterReady(dc *cassdcapi.CassandraDatacenter) bool {
	return dc.GetConditionStatus(cassdcapi.DatacenterReady) == corev1.ConditionTrue && dc.Status.CassandraOperatorProgress == cassdcapi.ProgressReady
}
```

The only reason why I extracted this comparison out of `DatacenterReady` is because there are two functions, `DatacenterStopped` and `DatacenterStopping` that differ only by the handling of `CassandraOperatorProgress`. 

Do you advocate instead for changing how `DatacenterReady` is implemented? I wanted to avoid this.

Reply:
I might be in favor of changing `DatacenterReady`. It doesn't have to be done in this PR (if it is done at all).

Reply:
OK I removed the if block and came back to something closer to what we have in main. We can then see later to change DatacenterReady, but it is called in many places, so definitely something we want to do in another PR.